src/config/sns.json

```diff
@@ -0,0 +1,41 @@
+{
+  "name": "SNS",
+  "displayName": "Simple Notification Service",
+  "rules": [
+    {
+      "aws": {
+        "logGroup": "sns/%{DATA:region_id}/%{DATA:account_id}/%{USERNAME:resource_id}%{GREEDYDATA:failure_suffix}"
+      },
+      "sources": [
+        {
+          "sourceType": "logs",
+          "source": "log_group",
+          "condition": "$prefix('sns/')"
+        }
+      ],
+      "attributes": [
+        {
+          "key": "aws.service",
+          "pattern": "'sns'"
+        },
+        {
+          "key": "aws.resource.id",
+          "pattern": "resource_id"
+        },
+        {
+          "key": "aws.arn",
+          "priority": 1,
+          "pattern": "format_arn('arn:{}:sns:{}:{}:{}', [partition, region, account_id, resource_id])"
+        },
+        {
+          "key": "severity",
+          "pattern": "if( failure_suffix == '', &'INFO', &'ERROR', @)"
+        },
```

Comment:
I thought we should get audit.result for those error logs, similiarly as for CloudTrail.
`\\\"ErrorCode\\\":\\\"AccessDenied\\\`
```
        {
          "key": "audit.result",
          "description": "If a field errorCode exists, then return Failure.{errorCode}, else return Succeeded",
          "pattern": "if( log_content.errorCode==null, &'Succeeded', &[`Failed`, log_content.errorCode] | [? @!=null ] | join('.', @) , @)"
        },
```
You would also need to add
`  "logContentParseAs": "json"`
for that, so I expect resistance here 😄 and PawelS is not here for me to ask and make sure when do we want this field filled and where not.

Reply:
From what I know we don't need this right now (use case is following: you select all logs with "ERROR" and then look at them on your own, that's all we need). But if you see some use cases or expect Pawel to see some, can you follow up this with Pawel?

Reply:
Followed up with Marcin, final decision was this should not be done, but not because "we don't need this right now". The actual reason was to keep "**audit.XXX**" fields reserved for "audit" logs, i**.e. logs about configuration/deployment changes**, and this one here seems to be "message accepted/or not" rather than "SNS settings changed successfully or not".